packages/contracts-bedrock/test/setup/ForkLive.s.sol

```diff
@@ -93,6 +94,23 @@ contract ForkLive is Deployer {
             IFaultDisputeGame(address(disputeGameFactory.gameImpls(GameTypes.PERMISSIONED_CANNON)));
         save("PermissionedDisputeGame", address(permissionedDisputeGame));
         save("PermissionedDelayedWETHProxy", address(permissionedDisputeGame.weth()));
+
+        // Now deploy the updated OPCM and implementations of the contracts
+        _deployNewImplementations();
+    }
+
+    /// @notice Etches a new Deploy.s.sol contract at a deterministic address, sets up the environment,
+    ///         and deploys new implementations with a "_NextVersion" suffix. This suffix is necessary to avoid
+    ///         naming collisions with the implementations saved above.
+    function _deployNewImplementations() internal {
+        Deploy deployNew = Deploy(address(uint160(uint256(keccak256(abi.encode("optimism.deploy.new"))))));
+        vm.etch(address(deployNew), vm.getDeployedCode("Deploy.s.sol:Deploy"));
+        vm.label(address(deployNew), "DeployNew");
+        vm.allowCheatcodes(address(deployNew));
+        vm.setEnv("CONTRACT_ADDRESSES_PATH", string.concat(vm.projectRoot(), "/deployments/1-deploy.json"));
+
+        deployNew.setUp();
+        deployNew.deployImplementations({ _isInterop: false, _suffix: "_NextVersion" });
     }
```

Comment:
I know we also talked about an approach where `deployImplementations` is create2 based, so we could have it do something like this:
- compute address for deploying a contract
- check if there's code there
- if code, contract is unchanged, skip deploy
- if no code, contract changed, execute deploy

Is my interpretation correct that this PR is an alternative solution to that one? The create2 based one feels cleaner and simpler to me though. Here we are further enshrining the `save` functionality by relying on it and I thought we wanted to move away from that?

Reply:
Resolving. There's a ticket for create2, and this was addressed by allowing addreses to be saved over.